lib/Array.cry

```diff
@@ -11,3 +11,47 @@ primitive arrayConstant : {a, b} b -> (Array a b)
 primitive arrayLookup : {a, b} (Array a b) -> a -> b
 primitive arrayUpdate : {a, b} (Array a b) -> a -> b -> (Array a b)
 
+/**
+ * Copy elements from the source array to the destination array.
+ *
+ * 'arrayCopy dest_arr dest_idx src_arr src_idx len' copies the
+ * elements from 'src_arr' at indices '[src_idx .. (src_idx + len)]' into
```

Comment:
Sorry to nitpick some more, but I think essentially all these ranges are intended to be half-open and we should probably use the Cryptol-style notation for that, which would make this `[src_idx ..< (src_idx + len)]`, and similar below.